rfd/0147-repo-fork-syncing.md

```diff
@@ -0,0 +1,60 @@
+---
+authors: Mike Jensen (mike.jensen@goteleport.com)
+state: draft
+---
+ 
+# RFD 144 - Automatic Repo Syncing
+ 
+## What
+ 
+This RFD describes how we can sync a repo continuously and in an automatic way.  This can be used to provide private copies of a repo.
+ 
+## Why
+ 
+We currently face a few frictions which could be addressed through a more robust repo sync mechanism:
+1. Our internal `teleport-private` fork used for security development must be manually updated
+2. Some tooling (Dependabot) does not allow scanning for non-default branches, necessitating manual efforts to manage our release branches
+3. Because much of the CI and scanning configuration is committed into the repo, we face noise from duplicate alerts when we use a repo for testing or as an internal fork
+ 
+## Details
+ 
+As such we need a solution that can provide the following:
+* Automatic and regular updates to a repo to ensure it's in sync
+* The ability to apply changes to the upstream repo (for example disable Actions that we don't want running on the repo copy, or enable additional scanning / actions)
+ 
+### Branch Structure
+ 
+* `main` - This will be the repos default branch.  This branch will contain the upstream history in addition to commits needed for the sync process or other custom changes on the repo copy
+* `sync/upstream-[UPSTREAM_BRANCH_NAME]` - This branch will be an exact copy of the upstream branch.  If there is a need to create a PR in this repo the history should be based off this rather than `main`
+* `sync/rebase` - This branch is where our custom changes will be committed.  It will be rebased on to the upstream changes before being committed into `main`
```

Comment:
I'm not sure I understand this rebase branch. We'll have many upstream branches that correspond to our release branches, but a single rebase branch?

Reply:
There will in total be 4 repo's:
* `teleport-private` (will contain the syncing of all `branch/v*` as well as `master`)
* `teleport-sec_scan-X` which there will be 3 of so that we can do security scanning on the repos default branch

Reply:
So after every major release you'll need to re-sync all of the teleport-sec_scan repos to the correct release branch? (-1 becomes v14, -2 becomes v13, etc)

Will that part be automated or done manually?

Reply:
I clarified the 4 repo process in the RFD, let me know if there are further questions.